go/worker/compute/executor/committee/node.go

```diff
@@ -601,6 +611,45 @@ func (n *Node) removeTxBatch(batch [][]byte) error {
 	return nil
 }
 
+func (n *Node) proposeTimeoutLocked() error {
+	if n.commonNode.CurrentBlock == nil {
+		return fmt.Errorf("executor: propose timeout error, nill block")
+	}
+	if n.commonNode.CurrentBlockHeight+n.proposerTimeout > n.commonNode.Height {
+		n.logger.Debug("executor: proposer timeout not reached yet",
+			"height", n.commonNode.Height,
+			"current_block_height", n.commonNode.CurrentBlockHeight,
+			"proposer_timeout", n.proposerTimeout,
+		)
+		return nil
+	}
+
+	timeout, err := commitment.SignProposerTimeout(n.commonNode.Identity.NodeSigner, n.commonNode.CurrentBlock.Header.Round)
+	if err != nil {
+		n.logger.Error("failed to sign txn scheduler timeout request",
+			"err", err,
+		)
+		return fmt.Errorf("failed to sign txn scheduler timeout request: %w", err)
+	}
+
+	tx := roothash.NewRequestProposerTimeoutTx(0, nil, n.commonNode.Runtime.ID(), *timeout)
+	go func() {
+		err := consensus.SignAndSubmitTx(n.roundCtx, n.commonNode.Consensus, n.commonNode.Identity.NodeSigner, tx)
+		switch err {
+		case nil:
+			n.logger.Info("executor timeout request finalized")
+		default:
+			n.logger.Error("failed to submit executor timeout request",
+				"err", err,
+			)
+		}
+	}()
+
+	n.transitionLocked(StateWaitingForTimeout{})
```

Comment:
I added it mostly to avoid re-proposing timeout after we do it once for the round, but i think it could be fine as well if we just stay in the WaitingForBatch state.

Reply:
Oh I see, yeah that makes sense, but maybe do it with a flag (that is cleared on the next block or when the timeout request succeeds/fails) instead of a new state.

Reply:
yeah that makes more sense, thanks!